### PR TITLE
planner: add some e2e test cases for outer join elimination

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #54057

Problem Summary: planner: add some e2e test cases for outer join elimination

### What changed and how does it work?

1. Add some e2e test cases for outer join elimination
2. Rename Schema's Keys and UniqueKeys to PKOrUK and NullableUK to be more clear.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: pkg/expression/schema.go [hunk 1]
```diff
@@ -46,10 +46,10 @@ func (ki KeyInfo) String() string {
 // Schema stands for the row schema and unique key information get from input.
 type Schema struct {
 	Columns []*Column
-	Keys    []KeyInfo
-	// UniqueKeys stores those unique indexes that allow null values, but Keys does not allow null values.
-	// since equivalence conditions can filter out null values, in this case a unique index with null values can be a Key.
-	UniqueKeys []KeyInfo
+	PKOrUK  []KeyInfo // this fields stores the primary key or unique key.
+	// NullableUK stores those unique indexes that allow null values, but Keys does not allow null values.
```
Comment (windtalker) [inline pkg/expression/schema.go:50]:
```suggestion
	// NullableUK stores those unique indexes that allow null values, but PKOrUK does not allow null values.
```

**Review by windtalker — APPROVED** [review]
others lgtm

**Pull request merged**